### PR TITLE
Adding Python Book Named as Pro Python Advanced coding techniques and tools By Marty Alchin

### PR DESCRIPTION
**Pull request by CyberSecByte**
## What does this PR do?
Add resource

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1933,6 +1933,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Porting to Python 3: An In-Depth Guide](http://python3porting.com) (2.6 - 2.x & 3.1 - 3.x)
 * [Practical Programming in Python](https://launchpadlibrarian.net/165489933/PracticalProgrammingPython2014.pdf) - Jeffrey Elkner (PDF)
 * [Practice Python Projects](https://learnbyexample.github.io/practice_python_projects/) - Sundeep Agarwal
+* [Pro Python Advanced coding techniques and tools](https://archive.org/download/python-books/Apress.Pro.Python.Advanced.Coding.Techniques.And.Tools.Jun.2010.ISBN.1430227575.pdf) - Marty Alchin (PDF)
```
[the remark this answers is not in the document]
Comment (CyberSecByte) [reply books/free-programming-books-langs.md:1936]: No this is a free book and available at archive.org

[the remark this answers is not in the document]
Comment (CyberSecByte) [reply books/free-programming-books-langs.md:1936]: also you can see this post https://searchworks.stanford.edu/view/13192700

**Comment by eshellman** [comment]
add the "archived" notation please.
We allow links to IA because they have a good DMCA process, so if springer cares about this book, they can easily get it removed. I doubt they will, because it's an obsolete edition and people will want the new one.

**Comment by CyberSecByte** [comment]
> Internet Archive

I added the archive tag and this it's available on https://archive.org/details/python-books/053f88d62b94d30b5e22278e906c5ee8/ since 2020 and i don't think it's a paid one and the owner will take a dmca action on it

**Comment by LuigiImVector** [comment]
Use the latest version of the book

**Pull request merged**